lib/paper_trail/record_trail.rb

```diff
@@ -296,7 +296,36 @@ def data_for_update
         data[:created_at] = @record.updated_at
       end
       if record_object_changes?
-        data[:object_changes] = recordable_object_changes
+        data[:object_changes] = recordable_object_changes(changes)
+      end
+      add_transaction_id_to(data)
+      merge_metadata_into(data)
+    end
+
+    def record_update_columns(changes)
+      if enabled?
+        versions_assoc = @record.send(@record.class.versions_association_name)
+        version = versions_assoc.create(data_for_update_columns(changes))
+        if version.errors.any?
+          log_version_errors(version, :update)
```

Comment:
Nice job including details like this error logging, and the `enabled?` check. :+1: